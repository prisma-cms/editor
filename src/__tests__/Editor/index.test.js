import React from 'react';
import { render } from '@testing-library/react'
 
const contentStr = `
  <div>
    Test content
    <ul>
      <li>Row 1</li>
      <li>Row 2</li>
    </ul>
  <div>  
`;


describe('Editor', () => {

  it('Render Editor with mocked generateRandomKey', () => {

    /**
     * Mock generateRandomKey();
     * But for production this not allowed, so if you use React.hydrate and SSR,
     * you'l get error for wrong data-offset-key
     */
    jest.mock('draft-js-android-fix/lib/generateRandomKey', () => {

      let count = 0;

      return () => {

        console.log("test generateRandomKey mock default");

        count++;

        return count;
      };
    });

    const Component = require('../../App').default;

    const editorKey = "test-editor";

    const tree = render(<Component
      editorKey={editorKey}
      value={contentStr}
    />)

    expect(tree.container).toMatchSnapshot()

    const node = tree.container.children[0]

    /**
     * Check editorKey
     */
    expect(node.querySelector("[data-editor]").attributes.getNamedItem("data-editor").value).toBe(editorKey);

    /**
     * Check content
     */
    // expect(node.querySelector("[data-text=true]")?.innerHTML).toBe(contentStr);
  });

})
