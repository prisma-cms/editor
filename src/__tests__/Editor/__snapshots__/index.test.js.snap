// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Render Editor with mocked generateRandomKey 1`] = `
<div>
  <div
    class="Component-root-1 "
  >
    <div
      class="DraftEditor-root"
    >
      <div
        class="DraftEditor-editorContainer"
      >
        <div
          aria-describedby="placeholder-test-editor"
          class="public-DraftEditor-content"
          contenteditable="false"
          spellcheck="true"
          style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
        >
          <div
            data-contents="true"
          >
            <div
              class=""
              data-block="true"
              data-editor="test-editor"
              data-offset-key="5-0-0"
            >
              <div
                class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                data-offset-key="5-0-0"
              >
                <span
                  data-offset-key="5-0-0"
                >
                  <span
                    data-text="true"
                  >
                        Test content     
                  </span>
                </span>
              </div>
            </div>
            <ul
              class="public-DraftStyleDefault-ul"
              data-offset-key="1-0-0"
            >
              <li
                class="public-DraftStyleDefault-unorderedListItem public-DraftStyleDefault-reset public-DraftStyleDefault-depth0 public-DraftStyleDefault-listLTR"
                data-block="true"
                data-editor="test-editor"
                data-offset-key="1-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="1-0-0"
                >
                  <span
                    data-offset-key="1-0-0"
                  >
                    <span
                      data-text="true"
                    >
                      Row 1
                    </span>
                  </span>
                </div>
              </li>
              <li
                class="public-DraftStyleDefault-unorderedListItem public-DraftStyleDefault-depth0 public-DraftStyleDefault-listLTR"
                data-block="true"
                data-editor="test-editor"
                data-offset-key="3-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="3-0-0"
                >
                  <span
                    data-offset-key="3-0-0"
                  >
                    <span
                      data-text="true"
                    >
                      Row 2
                    </span>
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
